### PR TITLE
Add docstrings to Tensorflow models module, and add more Tensorflow documentation in topics section

### LINKED ISSUE #111: Update Integrating Loss Functions topic to include TensorFlow
This section needs  to be updated to include guidance for TensorFlow.
https://backwardcompatibilityml.readthedocs.io/en/latest/backwardcompatibilityml/topics/integrating_loss_functions.html

### LINKED ISSUE #112: TensorFlow docstrings are not auto-generating documentation
The docstrings are present for all of the classes in the tensorflow namespace, but the documentation is not being auto-generated. See:
https://backwardcompatibilityml.readthedocs.io/en/v1.4.0/source/backwardcompatibilityml.tensorflow.html

### PR DESCRIPTION
**Pull request by ilmarinen**
This PR adds more documentation around the Tensorflow API for BackwardCompatibilityML.

### CODE REVIEW

**Review by jlema — APPROVED** [review]

**Review by jlema — APPROVED** [review]

File: backwardcompatibilityml/tensorflow/helpers.py [hunk 1]
```diff
@@ -2,6 +2,44 @@
...
+            h2,
+            training_set=ds_train,
+            testing_set=ds_test,
+            epochs=6,
+            bc_loss=bc_loss,
+            optimizer=optimizer)
+
+    Args:
+        h2: A Tensorflow model that we want to train using backward compatibility.
+        training_set: The training set to use to train our model.
```
Comment (KathleenW1) [inline backwardcompatibilityml/tensorflow/helpers.py:33]: Change to "The training set for our model."

File: backwardcompatibilityml/tensorflow/helpers.py [hunk 2]
```diff
@@ -2,6 +2,44 @@
...
+        testing_set: The testing set to use to validate our model.
```
Comment (KathleenW1) [inline backwardcompatibilityml/tensorflow/helpers.py:34]: Change to "The testing set for validating our model."

File: backwardcompatibilityml/tensorflow/models.py [hunk 3]
```diff
@@ -5,6 +5,40 @@
 
 
 class BCNewErrorCompatibilityModel(tf.keras.models.Sequential):
+    """
+    BackwardCompatibility base model for Tensorflow
+
+    One may create a new Tensorflow model by subclassing
```
Comment (KathleenW1) [inline backwardcompatibilityml/tensorflow/models.py:11]: Change instances of "One" to "You." ( per Microsoft style)   "You may create ...

File: backwardcompatibilityml/tensorflow/models.py [hunk 4]
```diff
@@ -5,6 +5,40 @@
...
+    their new model h2 from this model.
```
Comment (KathleenW1) [inline backwardcompatibilityml/tensorflow/models.py:12]: Change "their" to "your"

File: backwardcompatibilityml/tensorflow/models.py [hunk 5]
```diff
@@ -5,6 +5,40 @@
...
+    This allows one to train or update a new model h2, using the
```
Comment (KathleenW1) [inline backwardcompatibilityml/tensorflow/models.py:13]: Change "one" to "you."

**Comment by ilmarinen** [comment]
@KathleenW1 Thank you! I made the changes you requested.

File: backwardcompatibilityml/tensorflow/models.py (hunk 5 above)
Comment (ilmarinen) [reply backwardcompatibilityml/tensorflow/models.py]: Done.

File: backwardcompatibilityml/tensorflow/models.py (hunk 4 above)
Comment (ilmarinen) [reply backwardcompatibilityml/tensorflow/models.py]: Done.

File: backwardcompatibilityml/tensorflow/models.py (hunk 3 above)
Comment (ilmarinen) [reply backwardcompatibilityml/tensorflow/models.py]: Done.

File: backwardcompatibilityml/tensorflow/helpers.py (hunk 2 above)
Comment (ilmarinen) [reply backwardcompatibilityml/tensorflow/helpers.py]: Done.

File: backwardcompatibilityml/tensorflow/helpers.py (hunk 1 above)
Comment (ilmarinen) [reply backwardcompatibilityml/tensorflow/helpers.py]: Done.

File: backwardcompatibilityml/tensorflow/models.py [hunk 6]
```diff
@@ -5,6 +5,40 @@
...
+    backward compatibility loss, with respect to an existing
+    model h1, using the Tensorflow fit method, h2.fit(...).
+
+    Assuming that you have a pre-trained model h1 and you
+    would like to create a new model h2 that you train
```
Comment (KathleenW1) [inline backwardcompatibilityml/tensorflow/models.py:18]: Change to:  "would like to create a new model h2 trained"

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst [hunk 7]
```diff
@@ -0,0 +1,93 @@
+.. _using_the_tensorflow_interface:
+
+The Tensorflow Interface
+=================================================================
+
+There are two ways in which one may use the notion of the Backward
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst:6]: Change "one" to "you"

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+Compatibility loss with respect to the Tensorflow framework.
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst:7]: Lowercase "compatibility."

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+- Creating and training a model subclassed from BCNewErrorCompatibilityModel
+- Training a general model h2 using Backward Compatibility Loss
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst:10]: Lowercase "backward compatibility loss"

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+The first method for achieving this allows one to leverage the existing
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst:12]: Change "one" to "you."

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+``h2.fit(...)`` method in order to train your model. However it requires
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst:13]: Comma after "however."

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+that ``h2`` was instantiated from a model subclassed from BCNewErrorCompatibilityModel.
+If you already had a model instantiated from a distinct model class, then
+you mayneed to go through soe effort to extract the layers you need
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst:16]: Space between "may" and "need."  Change "through soe" to "to some"

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+and wrap them within a new model class subclassed from BCNewErrorCompatibilityModel.
+
+On the other hand, the second method places no restrictions on the architecture
+of ``h2``. However we will be unable to use the existing Tensorflow
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst:20]: Comma after "however."

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+it using ``tf_helpers.bc_fit``.
+
+We go into the details of both methods below.
+
+
+Creating and training a model subclassed from BCNewErrorCompatibilityModel
+---------------------------------------------------------------------------
+
+Assuming that you have a pre-trained model ``h1`` and that you want to
+create a new model ``h2`` which is to be trained using the
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst:31]: Change "which" to "that."

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+Backward Compatibility loss, with respect to ``h1`` for some value of
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst:32]: Lowercase "backward compatibility."

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+    )
+
+    h2.fit(
+        dataset_train,
+        epochs=6,
+        validation_data=dataset_test,
+    )
+
+
+An example notebook which walks you through a working example may be found at
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst:58]: Change "which" to "that."

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+    tf_helpers.bc_fit(
+        h2,
+        training_set=ds_train,
+        testing_set=ds_test,
+        epochs=6,
+        bc_loss=bc_loss,
+        optimizer=optimizer)
+
+
+An example notebook which walks you through a working example may be found at
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst:92]: Change "which" to "that."

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst [hunk 18]
```diff
@@ -0,0 +1,93 @@
...
+        epochs=6,
+        validation_data=dataset_test,
+    )
+
+
+An example notebook which walks you through a working example may be found at
+``./examples/tensorflow-MNIST`` from the BackwardCompatibilityML project root.
+
+
+Training a general model h2 using Backward Compatibility Loss
```
Comment (KathleenW1) [inline docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst:62]: Lowercase "backward compatibility loss."

**Review by KathleenW1 — CHANGES_REQUESTED** [review]
Edits indicated via in-line comments.

File: backwardcompatibilityml/tensorflow/models.py (hunk 6 above)
Comment (ilmarinen) [reply backwardcompatibilityml/tensorflow/models.py]: Done.

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst (hunk 7 above)
Comment (ilmarinen) [reply docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst]: Done.

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst (hunk 8 above)
Comment (ilmarinen) [reply docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst]: Done.

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst (hunk 9 above)
Comment (ilmarinen) [reply docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst]: Done.

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst (hunk 10 above)
Comment (ilmarinen) [reply docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst]: Done.

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst (hunk 11 above)
Comment (ilmarinen) [reply docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst]: Done

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst (hunk 12 above)
Comment (ilmarinen) [reply docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst]: Done.

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst (hunk 13 above)
Comment (ilmarinen) [reply docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst]: Done.

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst (hunk 14 above)
Comment (ilmarinen) [reply docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst]: Done.

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst (hunk 15 above)
Comment (ilmarinen) [reply docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst]: Done.

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst (hunk 16 above)
Comment (ilmarinen) [reply docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst]: Done.

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst (hunk 17 above)
Comment (ilmarinen) [reply docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst]: Done.

File: docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst (hunk 18 above)
Comment (ilmarinen) [reply docs/backwardcompatibilityml/topics/using_the_tensorflow_interface.rst]: Done.

**Comment by ilmarinen** [comment]
@KathleenW1 Made the latest round of fixes.

**Review by KathleenW1 — APPROVED** [review]

**Review by nushib — APPROVED** [review]
